src/staging/src/components/park/parkFacility.js

```diff
@@ -11,61 +12,111 @@ import StaticIcon from "./staticIcon"
 import "../../styles/cmsSnippets/parkInfoPage.scss"
 
 export default function ParkFacility({ data }) {
-  const facilityData = data
-  const [expanded, setExpanded] = useState(
-    Array(facilityData.length).fill(false)
+  const [facilityData, setFacilityData] = useState(
+    JSON.parse(JSON.stringify(data)) // deep copy
   )
+  const [expanded, setExpanded] = useState(Array(data.length).fill(false))
+  const [numFacilities, setNumFacilities] = useState(0)
 
-  if (facilityData.length === 0) return null
+  const toggleExpand = useCallback(
+    index => {
+      expanded[index] = !expanded[index]
+      setExpanded([...expanded])
+    },
+    [expanded]
+  )
+  //const [isInitial, setIsInitial] = useState(true)
+
+  const checkHash = useCallback(() => {
+    // Check hash in url
+    // if we find a matching facilityCode, open that facility accordion
+    // TODO setup useEffect etc.
+    let h = ""
+
+    if (typeof window !== "undefined") {
```

Comment:
Recommend using useLocation from reach router (used by Gatsby) instead of checking window:
```
import { useLocation } from "@reach/router"

...
const location = useLocation()
const hash = location.hash
```

Should update without the need for a manual event listener

Reply:
I could find a simple way to use this approach without causing a lot of refactoring and/or "too many rerender" errors. Leaving as is.